### PR TITLE
#65 code quality implementations

### PR DESCRIPTION
**Pull request by refatK**
**NOTE: don't merge yet**

#65

Adding code quality analysis tools to our project. I added:

- **Sonarcloud**: catches code smells. Link: https://sonarcloud.io/dashboard?id=Landry333_Big-Owl
- **ReviewDog**: will give warnings from AndroidLint directly on our PRs. Should help keeping things clean

I added an issue in this commit that should have been caught:
https://github.com/Landry333/Big-Owl/commit/e263b33eee9bd8c28ccfe098b732dc82f63fa8ac

**Please answer my questions I asked here:**
https://github.com/Landry333/Big-Owl/pull/156#issuecomment-724287169

Should also be shown on PR:
image

### CODE REVIEW

**Comment by refatK** [comment]
Reviewdog provides 3 types of reporter. Which should we use? We can also set different ones depending on the action:

1. [GitHub Checks (-reporter=github-pr-check)](https://github.com/reviewdog/reviewdog#reporter-github-checks--reportergithub-pr-check) (I think this only checks the changes within the PR) ([example](https://github.com/reviewdog/reviewdog/pull/275/files#annotation_6177941961779419))

2. [GitHub Checks (-reporter=github-check)](https://github.com/reviewdog/reviewdog#reporter-github-checks--reportergithub-check) (currently set, shows all problems in code as of commit) ([example](https://github.com/reviewdog/reviewdog/pull/364/files) [also the current state of the PR at this point of time]) (we can also setup a [badge](https://github.com/reviewdog/reviewdog#reviewdog-badge-) with this one)


3. [GitHub PullRequest review comment (-reporter=github-pr-review)](https://github.com/reviewdog/reviewdog#reporter-github-pullrequest-review-comment--reportergithub-pr-review) (will be like a PR review instead) ([example](https://github.com/reviewdog/reviewdog/pull/24/files))


Which ones should we use. We can even use multiple based on what we are doing (ex: only do on staging merge, only on PR, ...)

Also I currently set it to be warning only. The CI will not fail the PR, it only warns you about the lint issues.\!

File: build.gradle [hunk 1]
```diff
@@ -6,8 +6,8 @@ buildscript {
         maven { url 'https://oss.jfrog.org/artifactory/oss-snapshot-local' }
     }
     dependencies {
-        //Do not update the build version
-        //If this version doens't work, go to gradle/wrapper/gradle-wrapper.properties and change distributionUrl gradle version to 6.1.1
+        // Do not update the build version (keep: classpath 'com.android.tools.build:gradle:4.0.2')
```
Comment (ShawnGuerra-Bautista) [inline build.gradle:9]: Does this enforce that this dependency must be present?

Also, if let's say we add 
`classpath 'com.android.tools.build:gradle:6.1.1'`
and we don't remove 4.0.2, does Gradle complain?

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -54,4 +54,5 @@
     values-round/strings.xml for round devices.
     -->
     <string name="hello_world">Hello Square World!</string>
+    <string name="unused_resource_for_review_dog_check_test">Hopefully reviewdog calls this out. Will be removed!</string>
```
Comment (ShawnGuerra-Bautista) [inline app/src/main/res/values/strings.xml:57]: Cool the lint does work and show bad code!

You can remove it.

**Review by ShawnGuerra-Bautista — CHANGES_REQUESTED** [review]
All the configuration looks good. Good job.

I know you already stated this in the code's comments, but just remember to remove the example that you provided for the lint.

**Comment by refatK** [comment]
Seems like Sonarcloud sets up reports for the changes within the branch too. Good stuff.

**Review by KhangZhengNg — APPROVED** [review]
This is totally awesome! thanks for adding it to our project!

**Review by ShawnGuerra-Bautista — APPROVED** [review]

**Pull request merged**